### PR TITLE
feat: integration with GoogleAnalytics

### PR DESCRIPTION
**Pull request by ShahAnuj2610**

### CODE REVIEW

File: content/docs/analytics/GoogleAnalytics.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    - concepts
+    - appbase
+    - google-analytics
+    - integrate
+sidebar: 'docs'
+---
+
+##Setting Up Google Analytics
+Even though ReactiveSearch provides an analytics tailored to your search, you might want to integrate your search into your existing analytics tools.
+The base library components doesn't provide a built-in widget to implement analytics with other providers.
```
Comment (siddharthlatest) [inline content/docs/analytics/GoogleAnalytics.md]: s/doesn't/don't (since components is plural)

File: content/docs/analytics/GoogleAnalytics.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Integrating with Google Analytics requires 2 steps:
+
+-   set up the Google Analytics library in your page
```
Comment (siddharthlatest) [inline content/docs/analytics/GoogleAnalytics.md]: in your `search app` (instead of page)

Comment (siddharthlatest) [reply content/docs/analytics/GoogleAnalytics.md]: Please link to their installation page.

File: content/docs/analytics/GoogleAnalytics.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+-   setup the search state change listener
+
+To set up Google Analytics, the best way is to actually [follow the reference guide](https://developers.google.com/analytics/devguides/collection/analyticsjs/).
+
+Once the GA library is installed on your website, look below how Google Analytics can be integrated with different libraries
```
Comment (siddharthlatest) [inline content/docs/analytics/GoogleAnalytics.md]: `look below how` should be `follow the steps below on how`

File: content/docs/analytics/GoogleAnalytics.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+##Integrating With Libraries
```
Comment (siddharthlatest) [inline content/docs/analytics/GoogleAnalytics.md]: Title should be: "Setup The Search State Change Listener" based on the two steps you highlighted above. If this isn't accurately worded, also change the referenced bullet point's text.

Comment (siddharthlatest) [reply content/docs/analytics/GoogleAnalytics.md]: Add the following line before starting the library specific sections.

"Setting the search state change listener depends on how your search app is setup. If you are using one of the appbase.io libraries, you can refer to the sections below on how you can do this."

File: content/docs/analytics/GoogleAnalytics.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+###ReactiveSearch
+Track any component's state using `StateProvider` component as below. Read more about `StateProvider` [here](https://docs.appbase.io/docs/reactivesearch/v3/advanced/stateprovider/):
```
Comment (siddharthlatest) [inline content/docs/analytics/GoogleAnalytics.md]: `StateProvider [here]` should be `StateProvider over here`.

File: content/docs/analytics/GoogleAnalytics.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+			if (prevState !== nextState) {
+				window.ga('set', 'page', `/?state=${JSON.stringify(nextState)}`);
+				window.ga('send', 'pageview');
+			}
+		}}
+	/>
+</ReactiveBase>
+```
+
+If you want to track only your `DataSearch` component, you can add the following. Read more about `DataSearch` [here](https://docs.appbase.io/docs/reactivesearch/v3/search/datasearch/):
```
Comment (siddharthlatest) [inline content/docs/analytics/GoogleAnalytics.md]: Rephrase: If you only want to track your searchbox and you are using the [`DataSearch`](link to doc) component for it, you can do the following.

File: content/docs/analytics/GoogleAnalytics.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+		onValueChange={value => {
+			window.ga('set', 'page', `/?query=${value}`);
+			window.ga('send', 'pageview');
+		}}
+	/>
+</ReactiveBase>
+```
+
+###React SearchBox
+Check the API docs for `React SearchBox` [here](https://docs.appbase.io/docs/reactivesearch/react-searchbox/apireference/).
```
Comment (siddharthlatest) [inline content/docs/analytics/GoogleAnalytics.md]: `React SearchBox [here]` becomes `React SearchBox over [here]`.

File: content/docs/analytics/GoogleAnalytics.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+	dataField={['original_title', 'original_title.search']}
+	dataField
+	onValueChange={(next, prev) => {
+		if (prev !== next) {
+			window.ga('set', 'page', `/?query=${next}`);
+			window.ga('send', 'pageview');
+		}
+	}}
+/>
+```
```
Comment (siddharthlatest) [inline content/docs/analytics/GoogleAnalytics.md:88]: @ShahAnuj2610 You're missing the section for Vue.JS ReactiveSearch.

**Review by siddharthlatest — CHANGES_REQUESTED** [review]
See the specific comments.

File: content/docs/analytics/GoogleAnalytics.md (hunk 2 above)
Comment (ShahAnuj2610) [reply content/docs/analytics/GoogleAnalytics.md]: it's linked on `L22`

**Review by siddharthlatest — APPROVED** [review]

**Pull request merged**